app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ReservationsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_donor
+
+
+  def ship
+    @reservation = Reservation.find(params[:id])
+
+    if @reservation.donor_id == current_user.id
+      @reservation.ship!
+      render json: @reservation
+    else
+      redirect_to root_path, alert: "You are not authorized to access that page."
```

Comment:
Not sure if we're doing JSON or HTTP here - if JSON, this should render a 403 error, if HTTP it should redirect instead of rendering on success.